### ISSUE TITLE
Kconfig throws errors

### ISSUE THREAD

**Issue by pscollins** [lkl/linux#108]
I see the following when trying to compile:

```
cd tools/lkl; make -j
scripts/kconfig/conf  --defconfig Kconfig
kernel/time/Kconfig:155:warning: range is invalid
fs/sysfs/Kconfig:1:error: recursive dependency detected!
fs/sysfs/Kconfig:1:	symbol SYSFS is selected by AT91_ADC
drivers/iio/adc/Kconfig:121:	symbol AT91_ADC depends on IIO
drivers/iio/Kconfig:5:	symbol IIO is selected by RTC_DRV_HID_SENSOR_TIME
drivers/rtc/Kconfig:1608:	symbol RTC_DRV_HID_SENSOR_TIME depends on USB_HID
drivers/hid/usbhid/Kconfig:4:	symbol USB_HID depends on USB
drivers/usb/Kconfig:41:	symbol USB is selected by MOUSE_APPLETOUCH
drivers/input/mouse/Kconfig:177:	symbol MOUSE_APPLETOUCH depends on INPUT
drivers/input/Kconfig:8:	symbol INPUT is selected by VT
drivers/tty/Kconfig:12:	symbol VT is selected by FB_STI
drivers/video/fbdev/Kconfig:670:	symbol FB_STI depends on FB
drivers/video/fbdev/Kconfig:5:	symbol FB is selected by DRM_KMS_FB_HELPER
drivers/gpu/drm/Kconfig:34:	symbol DRM_KMS_FB_HELPER is selected by DRM_VIRTIO_GPU
drivers/gpu/drm/virtio/Kconfig:1:	symbol DRM_VIRTIO_GPU depends on VIRTIO
drivers/virtio/Kconfig:1:	symbol VIRTIO is selected by INTEL_MIC_CARD
drivers/misc/mic/Kconfig:57:	symbol INTEL_MIC_CARD depends on 64BIT
arch/lkl/Kconfig:35:	symbol 64BIT is selected by LKL
arch/lkl/Kconfig:1:	symbol LKL depends on AUDITSYSCALL
init/Kconfig:308:	symbol AUDITSYSCALL depends on AUDIT
init/Kconfig:296:	symbol AUDIT is selected by SECURITY_APPARMOR
security/apparmor/Kconfig:1:	symbol SECURITY_APPARMOR depends on SECURITY
security/Kconfig:21:	symbol SECURITY depends on SYSFS
```

Is this harmless, or does it matter?

**Comment by opurdila** [comment]
It looks harmless, but we should fix it nonetheless. 

I think it was introduced when we pulled in the whole drivers/Kconfig. @phhusson is there something that I forgot to merge?

**Comment by phhusson** [comment]
Hum no, I hadn't noticed this.
Trying to fix it raises many other such problems (next one is about VGA_CONSOLE)

The thing is that it says "depends on", but it's rather "depends on NOT" which actually breaks the loop rather than propagating it, so it sounds like a false positive to me

**Comment by pscollins** [comment]
@phhusson Yes, definitely. Surprisingly, I can't find any other examples of this kind of issue, so I'm not sure what the normal way to address it is.

Would it be worth removing some of the "negative dependencies" for the sake of quieting down the error message? The following builds with no warnings (other than invalid range):

    depends on !SMP && !MMU && !COREDUMP && !SECCOMP && !UPROBES && !COMPAT && !USER_RETURN_NOTIFIER

Or is preventing LKL from being compiled with bad options more important?

**Comment by phhusson** [comment]
My guess would be that there are options that must be ticked, and some that must NOT be ticked, which is already done by defconfig
So having negative dependencies for that is useless if it can be expressed in defconfig

**Comment by opurdila** [comment]
Closing as it is fixed by #128
